frontend/src/metabase/components/form/FormField.jsx

```diff
@@ -1,100 +1,102 @@
+    PropTypes.node,
+  ]),
+  className: PropTypes.string,
+};
+
+const ALL_DOT_CHARS = /\./g;
+
+function FormField(props) {
+  const {
+    className,
+    formField,
+    title = formField && formField.title,
+    description = formField && formField.description,
+    info = formField && formField.info,
+    hidden = formField && (formField.hidden || formField.type === "hidden"),
+    horizontal = formField &&
+      (formField.horizontal || formField.type === "boolean"),
+    children,
+  } = props;
+
+  if (hidden) {
+    return null;
+  }
+
+  let { name, error, visited, active } = {
+    ...(props.field || {}),
+    ...props,
   };
 
-  render() {
-    const {
-      className,
-      formField,
-      title = formField && formField.title,
-      description = formField && formField.description,
-      info = formField && formField.info,
-      hidden = formField &&
-        (formField.hidden != null
-          ? formField.hidden
-          : formField.type === "hidden"),
-      horizontal = formField &&
-        (formField.horizontal != null
-          ? formField.horizontal
-          : formField.type === "boolean"),
-      children,
-    } = this.props;
-
-    if (hidden) {
-      return null;
-    }
-
-    let { name, error, visited, active } = {
-      ...(this.props.field || {}),
-      ...this.props,
-    };
-
-    if (visited === false || active === true) {
-      // if the field hasn't been visited or is currently active then don't show the error
-      error = null;
-    }
-
-    return (
-      <div
-        className={cx("Form-field", className, {
-          "Form--fieldError": !!error,
-          flex: horizontal,
-        })}
-        id={`formField-${name.replace(/\./g, "-")}`}
-      >
-        {(title || description) && (
-          <div>
-            <FieldRow>
-              {title && (
-                <Label
-                  className={cx("Form-label", { "mr-auto": horizontal })}
-                  htmlFor={name}
-                  id={`${name}-label`}
-                >
-                  {title}
-                  {error && <span className="text-error">: {error}</span>}
-                </Label>
-              )}
-              {info && (
-                <Tooltip tooltip={info}>
-                  <InfoIcon />
-                </Tooltip>
-              )}
-            </FieldRow>
-            {description && <div className="mb1">{description}</div>}
-          </div>
-        )}
-        <div className={cx("flex-no-shrink", { "ml-auto": horizontal })}>
-          {children}
-        </div>
-      </div>
-    );
+  const formFieldId = `formField-${name.replace(ALL_DOT_CHARS, "-")}`;
+
+  if (!visited || active) {
+    // if the field hasn't been visited or is currently active then don't show the error
+    error = null;
   }
+
+  const rootClassNames = cx("Form-field", className, {
+    "Form--fieldError": !!error,
+    flex: horizontal,
+  });
+
+  return (
+    <div id={formFieldId} className={rootClassNames}>
```

Comment:
As we adopt styled components and grow more confident about our separation of concerns, I start to think:

what if we start making styled-components out of this 

```<div id={formFieldId} className={rootClassNames}>```

and all the logic to determine classes and styles is moved to the *.styled.js file?

Reply:
Good point, don't have a strong opinion on that for now